lib/rack/test.rb

```diff
@@ -194,8 +194,13 @@ def follow_redirect!
           else
             [:get, {}]
           end
+
+        # Compute the next location by appending the location header with the
+        # last request, as per https://tools.ietf.org/html/rfc7231#section-7.1.2
+        next_location = URI.parse(last_request.url) + URI.parse(last_response['Location'])
```

Comment:
I thought about this a bit more: does this work if `Location` is an absolute URL? It _feels_ like it wouldn't. So please add a test case for that, and if it works => fine, but if not I think we'll need to adjust the implementation a bit.

Reply:
When you add two absolute locations the RHS is returned. That's the behaviour of URI RFC.

Reply:
```
2.5.0 :001 > require 'uri'
 => false 
2.5.0 :002 > URI.parse("http://www.google.com") + URI.parse("http://www.yahoo.com")
 => #<URI::HTTP http://www.yahoo.com> 
2.5.0 :003 > 
```

Reply:
OK, sounds great! Would you mind adding a test that documents these semantics? You could even write a comment saying that `'# Adding two absolute locations returns the right-hand location` or something to the like.

Reply:
Ping @ioquatix